src/collective/cover/content.py

```diff
@@ -148,6 +149,26 @@ def get_referenced_objects(self):
                 refs |= getObjectsFromLinks(self, links)
         return refs
 
+    def purge_deleted_tiles(self):
+        """Check if there are any annotation of deleted tiles and destroy it.
+
+           Cover tiles are Browser Views, and data are saved in annotations.
+           When delete a tile, the annotations are not deleted, it keeps making
+           Cover object bigger;
+           This method purge all tile annotation data for tiles where there are
+           no reference in layout.
+        """
+        prefixes = ('plone.tiles.data', 'plone.tiles.configuration', 'plone.tiles.permission')
+        layout_tiles = self.list_tiles()
+        annotations = IAnnotations(self)
```

Comment:
this has to be divided in a two-step process: on plone.tiles persistent [tiles data can be stored on a storage different of annotations](https://github.com/plone/plone.tiles/blob/2.0.0b3/plone/tiles/data.py#L134-L136), that means it's safer to use the data manager API to remove the data from the tile; our annotations (configuration and permission) can be removed directly, but we will need to test them at least on the base tile at low level to guarantee it's working.

Reply:
I don't understand what you mean.. [storage variable is an alias to annotation property](https://github.com/plone/plone.tiles/blob/2.0.0b3/plone/tiles/data.py#L112-L114)

Reply:
I mean you have to delete the annotation using the tile manager.

Reply:
didn't work, tile collection test breaks because I don't know the type of the deleted tile to instantiate the correct type..